### PR TITLE
Feature: Press 'n' to go to next exercise

### PR DESCRIPTION
**Pull request by preshetin**
Thanks for such cool exercises! 

One thing that feels missing to me is the ability to quickly run the next exercise. 

<details>
  <summary> Click to see demo </summary>

next-exercise-demo  
</details>

Few notes on the implementation:
- I wrapped existing code in `runExercise` function. To improve readability, I did not make indentation of the code inside that function. So that Git does not mark those changes.
- Most of logic related to reading 'n' buttons is taken from https://stackoverflow.com/a/12506613/4990125

  
TODO:
- [x] It currently does not handle edge cases. So than when a user finishes last exercise and presses 'n' it will break things

### CODE REVIEW

**Comment by mattpocock** [comment]
This is wonderful! What a great addition. Could you:

1. Make the 'n' go to the next _solution_, not the next exercise
2. In the terminal, conditionally log 'Press 'n' to see the solution' or 'Press 'n' to go to exercise 09'
3. Make sure that on the final exercise, nothing is logged.

**Comment by preshetin** [comment]
Sounds good @mattpocock. Yes, I can do it. Hopefully on Sunday.

As a user, I did not even notice I could run `yarn solution 08`. My way of walking through the steps was like this:
1. Run the exercise in terminal.
2. Fix exercise code in text editor.
3. Make sure terminal output become "green".
4. Check in text editor for solutions.
5. Go to next exercise.


I'll look more closely into `yarn solution 08` program

**Comment by preshetin** [comment]
> 1. Make the 'n' go to the next _solution_, not the next exercise
> 2. In the terminal, conditionally log 'Press 'n' to see the solution' or 'Press 'n' to go to exercise 09'
> 3. Make sure that on the final exercise, nothing is logged.

Hey @mattpocock - it's done. You may check out to this PR branch and see how it feels.

There's extra feature included. When you press 'p' you go back to **p**revious step. So you can walk through tutorial with 'n' and 'p' buttons!

There is one bug (or feature?) though. When a user walks though a few steps and then decides to changes a file from previous exercise, the terminal 'refreshes' into that exercise. So it looks like the subprocesses from past exercises do not get killed. 

I can work on that bug also. Just want to hear some feedback on what's already done

File: scripts/exercise.js [hunk 1]
```diff
@@ -50,8 +57,85 @@ chokidar.watch(exerciseFile).on("all", (event, path) => {
     execSync(`tsc ${exerciseFile} --noEmit --strict`, {
       stdio: "inherit",
     });
-    console.log("Typecheck complete. You finished the exercise!");
+    console.log(`Typecheck complete. You finished the ${exercise} exercise!`);
+  
+    if (isEndOfTutorial(exercise, isSolution, pathIndicator)) {
+      console.log(`\n🎉 Cograts! You've reached to the end of this tutorial! 🎉  \n`);
```
Comment (jhonnymoreira) [inline scripts/exercise.js:63]:
```suggestion
      console.log(`\n🎉 Congrats! You've reached the end of this tutorial! 🎉  \n`);
```

Comment (preshetin) [reply scripts/exercise.js]: Nice catch @jhonnymoreira

Comment (jhonnymoreira) [reply scripts/exercise.js]: Nice work, @preshetin! 🤜🏻 🤛🏻

Comment (preshetin) [reply scripts/exercise.js]: Cool @jhonnymoreira , I wonder if you had a chance to try it on your local machine. I would really appreciate some validation

**Comment by jeff-fe** [comment]
I try this feature, it's great！🎉

**Comment by preshetin** [comment]
That bug with subprocess memory leak appeared to be quite nasty. Working on it.

**Comment by preshetin** [comment]
To fix that bug, I had to completely rewrite this feature. 

I had to introduce a brand new command `yarn tutorial 01`. This command listens for user input like 'n' or 'p' keypress and runs `yarn exercise XX` or `yarn solution XX` subprocesses. Before running new subprocess, it kills the old one. 

The commands `yarn exercice 03` and `yarn solution 03` work as usual. New  code I added to `scripts/exercise.js` gets executed only if this process is child. So when a user runs manually `yarn exercise 03` this additional code gets skipped.

I wonder @mattpocock if you are okay with intoducing this new `yarn tutorial` command. If yes, I can polish things like adding edge cases and updating the docs.

Feel free to test it

**Review by vaibhav4516 — COMMENTED** [review]
This is a great feature and it ease the work of the learner

**Comment by mattpocock** [comment]
@preshetin This is a very cool feature, but I think I'm not going to merge it. Here's why:

The intent of this repo was a sketch of my plans for a TS workshop. The workshop itself will eventually be hosted in a UI using Stackblitz, meaning that next/previous won't be necessary because users will navigate using the UI.

If I accept this PR, I'll need to ensure this code continues to work - including across versions of Node, and on all machines. I'm happy with the experience learners have currently. While this _is_ an improvement, I don't think it would be a _net_ improvement given that I'd need to care for and maintain this code.

Sorry for not coming to this conclusion earlier.
